connection.js

```diff
@@ -1184,19 +1184,14 @@ class Connection {
 
         // Validate source/destination IP
         /*eslint no-fallthrough: 0 */
-        switch (proto) {
-            case 'TCP4':
-                if (ipaddr.IPv4.isValid(src_ip) && ipaddr.IPv4.isValid(dst_ip)) {
-                    break;
-                }
-            case 'TCP6':
-                if (ipaddr.IPv6.isValid(src_ip) && ipaddr.IPv6.isValid(dst_ip)) {
-                    break;
-                }
-            // case 'UNKNOWN':
-            default:
-                this.respond(421, 'Invalid PROXY format');
-                return this.disconnect();
+        // TODO: verify replacing the switch-case here worked
+        if (proto == 'TCP4' && !ipaddr.IPv4.isValid(src_ip) && !ipaddr.IPv4.isValid(dst_ip)) {
```

Comment:
I'm pretty sure the logic here is wrong. Here I am again harping on legibility. This code is now harder to grok than the switch/case. Please revert.

Reply:
@msimerson Yeah, this was my own doing after trying to resolve a switch-case confusion error by SonarLint.  I went back over it, and updated the logic for clarity. In the old function, the switch-case fallthrough test was being disabled, and then it relied on a fallthrough to deal with unknown connection events. I added comments & an unknown connection entry: I think this will meet your readability needs, and avoid that ESLint hack.

Reply:
It doesn't meet my legibility concerns. If it did, it would be obvious to the casual reader that you changed the logic here and inadvertently broke it.

Reply:
Reverted, with comments to improve readability.